src/diffusers/quantizers/quantization_config.py

```diff
@@ -389,3 +392,270 @@ def to_diff_dict(self) -> Dict[str, Any]:
+                "float8sq": float8_static_activation_float8_weight,
+                # For fpx, only x <= 8 is supported by default. Other dtypes can be explored by users directly
+                # fpx weight + bfloat16/float16 activation
+                **generate_fpx_quantization_types(3),
+                **generate_fpx_quantization_types(4),
+                **generate_fpx_quantization_types(5),
+                **generate_fpx_quantization_types(6),
+                **generate_fpx_quantization_types(7),
+                # ===== Errors out with "torch.cat(): expected a non-empty list of Tensors" =====
+                # **generate_fpx_quantization_types(8),
+            }
+
+            UINTX_TO_DTYPE = {
+                1: torch.uint1,
+                2: torch.uint2,
+                3: torch.uint3,
+                4: torch.uint4,
+                5: torch.uint5,
+                6: torch.uint6,
+                7: torch.uint7,
+                # 8: torch.uint8,
+            }
+
+            def generate_uintx_quantization_types(bits: int):
+                types = {}
+                types[f"uint{bits}"] = partial(uintx_weight_only, dtype=UINTX_TO_DTYPE[bits])
+                types[f"uint{bits}wo"] = partial(uintx_weight_only, dtype=UINTX_TO_DTYPE[bits])
+                return types
+
+            UINTX_QUANTIZATION_DTYPES = {
+                "uintx": uintx_weight_only,
+                "uintx_weight_only": uintx_weight_only,
+                **generate_uintx_quantization_types(1),
+                **generate_uintx_quantization_types(2),
+                **generate_uintx_quantization_types(3),
+                **generate_uintx_quantization_types(4),
+                **generate_uintx_quantization_types(5),
+                **generate_uintx_quantization_types(6),
+                **generate_uintx_quantization_types(7),
+                # **generate_uintx_quantization_types(8),  # uint8 quantization is not supported
+            }
+
+            SHORTHAND_QUANTIZATION_TYPES = {
+                "int_a16w4": int4_weight_only,
+                "int_a8w4": int8_dynamic_activation_int4_weight,
+                "int_a16w8": int8_weight_only,
+                "int_a8w8": int8_dynamic_activation_int8_weight,
+                "uint_a16w1": partial(uintx_weight_only, dtype=torch.uint1),
+                "uint_a16w2": partial(uintx_weight_only, dtype=torch.uint2),
+                "uint_a16w3": partial(uintx_weight_only, dtype=torch.uint3),
+                "uint_a16w4": partial(uintx_weight_only, dtype=torch.uint4),
+                "uint_a16w5": partial(uintx_weight_only, dtype=torch.uint5),
+                "uint_a16w6": partial(uintx_weight_only, dtype=torch.uint6),
+                "uint_a16w7": partial(uintx_weight_only, dtype=torch.uint7),
+                # "uint_a16w8": partial(uintx_weight_only, dtype=torch.uint8),  # uint8 quantization is not supported
```

Comment:
Okay, I remove some redundant ones. I think we should support the following redundancy though:
- full function names of torchao for customization of kwargs by end user (for example, `int8_weight_only`.
- commonly used shorthands like `int8dq` and `int4wo`
- commonly used communication/documentation shorthands (`{base_dtype}_axwy`). 

I don't think it will be confusing because it is expected they know a little bit about torchao (and therefore their documentation notation of axwy) for the shorthands. Other than that, the full function names and `wo` and `dq` suffixes should not be too difficult to document well. Let me know if you think otherwise and want this removed

Reply:
I am okay with the current changes and the reasoning you provided. Thanks!